src/components/Album/Album.js

```diff
@@ -1,39 +1,90 @@
-import React, { useState, useEffect, useContext } from "react"
-import { useParams } from "react-router-dom"
-import './Album.css'
-import Form from '../Form/Form'
-import { getAlbumsByMasterId } from "../../api"
-import UserContext from "../UserContext/UserContext"
+import React, { useState, useEffect, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import { getAlbumDetails } from '../../api';
+import Form from '../Form/Form';
+import './Album.css';
+import UserContext from '../UserContext/UserContext';
 
 function Album() {
-  const {setCurrentUser} = useContext(UserContext)
-  const [album, setAlbum] = useState(null)
-  const [isLoading, setIsLoading] = useState(true)
-  const [modal, setModal] = useState(false)
-  
-  const {id} = useParams()
-  
-  const showModal = () => {
-    modal ? setModal(false) : setModal(true)
-  }
+  const { id } = useParams();
+  const [albumDetails, setAlbumDetails] = useState(null);
+  const [isLoading, setLoading] = useState(true);
+  const [modal, setModal] = useState(false);
+  const {currentUser, setCurrentUser} = useContext(UserContext) 
 
   useEffect(() => {
-    getAlbumsByMasterId(id)
-      .then(res => {
-        setAlbum(res)
-        setIsLoading(false)
+    setLoading(true);
+    getAlbumDetails(id)
+      .then(data => {
+        setAlbumDetails(data);
+        setLoading(false);
+      })
+      .catch(error => {
+        console.error('Error fetching album details:', error);
       })
-  }, [])
-
-  
-  if(!isLoading) {
-    return (
-      <>
-       <p>{id} hello</p>
-       {!modal && <button onClick={showModal}>Add a journal entry</button>}
-       {modal && <Form {...album} setCurrentUser={setCurrentUser} showModal={showModal}/>}
-      </>
-    )
-  } 
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [id]);
+
+  const handleAddToCollections = () => {
+    const { id, title, artist, coverImg } = albumDetails
```

Comment:
id isn't actually in albumDetails, we already have id from use params no need to reassign it, causing collections mapping to break

Reply:
i will remove the id property from handleAddToCollections- hopefully this will help

Reply:
so i removed id from this variable , and kept masterId in the newAlbum variable
This seems to fix the bug on my end , to where i can search an album -click on it- add it to collections- then go back to my collections tab and click on that same album to see Album details


![Screen Shot 2023-07-30 at 12 39 40 PM](https://github.com/mbenfowler/sound-stash/assets/76410246/7297d234-4c00-42b1-9527-98c83e3bf1b2)